lib/Doctrine/DBAL/Abstraction/V3Result.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Doctrine\DBAL\Abstraction;
+
+use Doctrine\DBAL\Driver\Exception;
+use Traversable;
+
+/**
+ * @see https://github.com/doctrine/dbal/blob/3.0.x/src/Result.php
+ */
+interface V3Result extends Result
```

Comment:
This should work. Otherwise, I'd rather use a more specific term like `ForwardCompatibility` and the corresponding namespace (see #4019) to simplify the cleanup.

Reply:
Seems a good idea to name it after the name used in 3.0.